src/NuGet.Services.AzureSearch/BatchPusherResult.cs

```diff
@@ -9,28 +9,37 @@ namespace NuGet.Services.AzureSearch
 {
     public class BatchPusherResult
     {
-        public BatchPusherResult() : this(new HashSet<string>())
+        public BatchPusherResult() : this(Array.Empty<string>())
         {
         }
 
-        public BatchPusherResult(HashSet<string> failedPackageIds)
+        public BatchPusherResult(IEnumerable<string> failedPackageIds)
         {
-            FailedPackageIds = failedPackageIds;
+            FailedPackageIds = failedPackageIds.ToHashSet(StringComparer.OrdinalIgnoreCase);
         }
 
         /// <summary>
         /// Package IDs that failed due to access condition on the version list.
         /// </summary>
         public HashSet<string> FailedPackageIds { get; }
 
-        public void EnsureNoFailures()
+        public bool Success => !FailedPackageIds.Any();
+
+        public void EnsureSuccess()
         {
-            if (FailedPackageIds.Any())
+            if (!Success)
             {
                 throw new InvalidOperationException(
                     "The index operations for the following package IDs failed due to version list concurrency: "
-                    + string.Join(", ", FailedPackageIds));
+                    + string.Join(", ", FailedPackageIds.OrderBy(x => x, StringComparer.OrdinalIgnoreCase)));
             }
         }
+
+        public BatchPusherResult Merge(BatchPusherResult other)
+        {
+            return new BatchPusherResult(FailedPackageIds
+                .Concat(other.FailedPackageIds)
+                .ToHashSet(FailedPackageIds.Comparer));
```

Comment:
Nit: is there a benefit to not be explicit here? As in, `.ToHashSet(StringComparer.OrdinalIgnoreCase));`? I don't feel strongly about this, feel free to skip if you prefer as-is.

Reply:
I was thinking a generic merge method should defer as much logic as possible and not "own" it's own logic if it's already defined elsewhere but sure I can change it

Reply:
I've thought many times about making a strong type for package ID but that's probably too prollerhead